yt/frontends/amrvac/data_structures.py

```diff
@@ -373,10 +374,14 @@ def _set_code_unit_attributes(self):
 
         # note: yt sets hydrogen mass equal to proton mass, amrvac doesn't.
         mp_cgs = self.quan(1.672621898e-24, "g")  # This value is taken from AstroPy
-        He_abundance = 0.1  # hardcoded parameter in AMRVAC
 
         # get self.length_unit if overrides are supplied, otherwise use default
         length_unit = getattr(self, "length_unit", self.quan(1, "cm"))
+        namelist = read_amrvac_namelist(self._parfiles)
+        if "mhd_list" in namelist:
+            He_abundance = namelist["mhd_list"].get("he_abundance", 0.1)
+        else:
+            He_abundance = 0.1  # default
```

Comment:
Inlining allows to keep a single source of truth for the default value
```suggestion
        He_abundance = namelist.get("mhd_list", {}).get("he_abundance", 0.1)
```